### PR TITLE
Use UnconfiguredProject.MakeRooted to compute project full file path

### PR DESCRIPTION
**Pull request by natidea**
The PR changes the computation of ProjectFileFullPath to use the CPS method `(UnconfiguredProject/PathHelper).MakeRooted`

/cc @dotnet/project-system

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs [hunk 1]
```diff
@@ -115,29 +120,62 @@ private static IVsReferenceItem GetReferenceItem(KeyValuePair<String, IImmutable
...
                 });
             }
+
             return referenceItems;
-        }        
+        }
+
+        private static string MakeRooted(string basePath, string path)
+        {
+            basePath = basePath.TrimEnd(Path.DirectorySeparatorChar);
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs]: `TrimEnd` takes an param []. You don't need to make 2 calls

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs [hunk 2]
```diff
@@ -115,29 +120,62 @@ private static IVsReferenceItem GetReferenceItem(KeyValuePair<String, IImmutable
...
+
+            // include package references with no 'Version' attribute
+            var packageProjects = packageReferenceItems.Where(p => !HasVersionAttribute(p.Value));
+            foreach (var packageProjectItem in packageProjects)
             {
-                var definingProjectDirectory = item.Properties.Item(DefiningProjectDirectoryProperty);
-                string fullPathFromDefining = definingProjectDirectory != null
-                    ? Path.Combine(definingProjectDirectory.Value, item.Name)
-                    : item.Name;
+                referenceItems.Add(GetReferenceItem(packageProjectItem));
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs]: can move `GetReferenceItem(packageProjectItem)` to the previous linq statement and `referenceItems.Add` can just add the resultant

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs [hunk 3]
```diff
@@ -115,29 +120,62 @@ private static IVsReferenceItem GetReferenceItem(KeyValuePair<String, IImmutable
...
+            }
 
-                string projectFileFullPath = Path.GetFullPath(fullPathFromDefining);
+            // compute project file full path property for each reference
+            foreach (ReferenceItem item in referenceItems)
+            {                
+                var definingProjectDirectory = item.Properties.Item(DefiningProjectDirectoryProperty);
+                string projectFileFullPath;
+                if (definingProjectDirectory != null)
```
Comment (basoundr) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs]:
```C#
var projectFileFullPath = definingProjectDirectory != null ?
                          MakeRooted(definingProjectDirectory.Value, item.Name) :
                          project.MakeRooted(item.Name); 
```

**Comment by davkean** [comment]
Is this relevant now that we've decided to make transitive Project References instead of reusing package references?

**Comment by natidea** [comment]
This change is no longer needed. I'll repurpose to just contain the changes to the computation of ProjectFileFullPath to use the CPS method `(UnconfiguredProject/PathHelper).MakeRooted`

**Comment by natidea** [comment]
I've limited the scope of this PR to using CPS MakeRooted APIs, and changed the title accordingly

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs [hunk 4]
```diff
@@ -1,141 +0,0 @@
-﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs]: Did you delete this file by mistake?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/VsItemList.cs [hunk 5]
```diff
@@ -34,28 +34,28 @@ public T Item(object index)
             return this[(int)index];
         }
 
-    public bool TryGetValue(string key, out T value)
-    {
-        // Until we have https://github.com/dotnet/corefx/issues/4690
+        public bool TryGetValue(string key, out T value)
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/VsItemList.cs]: Indentation got messed up?

Comment (natidea) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/VsItemList.cs]: Yes, this fixes the indentation from an earlier change.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs (hunk 4 above)
Comment (natidea) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs]: VS merge messed up here. I've fixed it.

**Pull request merged**